File: app/src/main/java/com/OdiousPanda/rainbow/API/RetrofitService.java
package com.OdiousPanda.rainbow.API;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitService {
    private static HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();

    private static OkHttpClient client = new OkHttpClient.Builder().addInterceptor(interceptor).build();

    private static Retrofit retrofitNearbySearch = new Retrofit.Builder()
            .baseUrl(Constant.GOOGLE_MAP_API_BASE_URL)
            .client(client)
            .addConverterFactory(GsonConverterFactory.create())
            .build();

    private static Retrofit retrofitWeather = new Retrofit.Builder()
            .baseUrl(Constant.BASE_URL)
            .client(client)
            .addConverterFactory(GsonConverterFactory.create())
            .build();

    private static Retrofit retrofitAQI = new Retrofit.Builder()
            .baseUrl(Constant.AQI_BASE_URL)
            .client(client)
            .addConverterFactory(GsonConverterFactory.create())
            .build();

    private static Retrofit retrofitUnSplash = new Retrofit.Builder()
            .baseUrl(Constant.UNPLASH_BASE_URL)
            .client(client)
            .addConverterFactory(GsonConverterFactory.create())
            .build();

    static {
        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
    }

    public static NearbySearchCall createNearbySearchCall() {
        return retrofitNearbySearch.create(NearbySearchCall.class);
    }

    public static WeatherCall createWeatherCall() {
        return retrofitWeather.create(WeatherCall.class);
    }

    public static AQICall createAQICall() {
        return retrofitAQI.create(AQICall.class);
    }

    public static UnsplashCall createUnsplashCall() {
        return retrofitUnSplash.create(UnsplashCall.class);
    }
}

File: app/src/main/java/com/odiousPanda/rainbowKt/apis/RetrofitService.kt
package com.odiousPanda.rainbowKt.apis

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitService {
    private val interceptor = HttpLoggingInterceptor()

    init {
        interceptor.level = HttpLoggingInterceptor.Level.BODY
    }

    private val client = OkHttpClient.Builder().addInterceptor(interceptor).build()

    private val retrofitWeather = Retrofit.Builder()
        .baseUrl(APIConstants.BASE_URL)
        .client(client)
        .addConverterFactory(GsonConverterFactory.create())
        .build()

    fun createWeatherCall(): WeatherCall {
        return retrofitWeather.create(WeatherCall::class.java)
    }

    private val retrofitNearbySearch = Retrofit.Builder()
        .baseUrl(APIConstants.GOOGLE_MAP_API_BASE_URL)
        .client(client)
        .addConverterFactory(GsonConverterFactory.create())
        .build()

    private val retrofitAQI = Retrofit.Builder()
        .baseUrl(APIConstants.AQI_BASE_URL)
        .client(client)
        .addConverterFactory(GsonConverterFactory.create())
        .build()

    private val retrofitUnSplash = Retrofit.Builder()
        .baseUrl(APIConstants.UNPLASH_BASE_URL)
        .client(client)
        .addConverterFactory(GsonConverterFactory.create())
        .build()

    fun createAQICall(): AQICall {
        return retrofitAQI.create(AQICall::class.java)
    }

    fun createUnsplashCall(): UnsplashCall {
        return retrofitUnSplash.create(UnsplashCall::class.java)
    }

    fun createNearbySearchCall(): NearbySearchCall {
        return retrofitNearbySearch.create(NearbySearchCall::class.java)
    }

}